dotnet/src/SemanticKernel.Abstractions/TemplateEngine/IPromptTemplateEngine.cs

```diff
@@ -14,14 +13,12 @@ namespace Microsoft.SemanticKernel.TemplateEngine;
 public interface IPromptTemplateEngine
 {
     /// <summary>
-    /// Given a prompt template string, extract all the blocks (text, variables, function calls)
+    /// Given a prompt template string, extract all the variable names
     /// </summary>
     /// <param name="templateText">Prompt template (see skprompt.txt files)</param>
-    /// <param name="validate">Whether to validate the blocks syntax, or just return the blocks found, which could contain invalid code</param>
-    /// <returns>A list of all the blocks, ie the template tokenized in text, variables and function calls</returns>
-    IList<Block> ExtractBlocks(
-        string? templateText,
-        bool validate = true);
+    /// <returns>A list of all the variable names</returns>
```

Comment:
nit: If we move towards something like `Guidance` -- we'll have both input and output variable names. Might want to consider that with renaming now (ExtractInputVariableNames or something)

Reply:
I'm removing this method completely